src/Provider/Provider.Web/Orchestrators/VacancyPreviewOrchestrator.cs

```diff
@@ -119,10 +127,18 @@ private async Task<SubmitVacancyResponse> SubmitActionAsync(Vacancy vacancy, Vac
 
             if (response.HasLegalEntityAgreement && response.HasProviderAgreement)
             {
-                var command = new SubmitVacancyCommand(vacancy.Id, user, OwnerType.Provider);
-
-                await _messaging.SendCommandAsync(command);
-                response.IsSubmitted = true;
+                if (hasProviderReviewPermission)
+                {
+                    var command = new ReviewVacancyCommand(vacancy.Id, user, OwnerType.Provider);
+                    await _messaging.SendCommandAsync(command);
+                    response.IsReviewed = true;
```

Comment:
should this property be `IsSentForReview`?

Reply:
Done